### ISSUE TITLE
TypeScript type definition

### ISSUE THREAD

**Issue by yo7** [mfrachet/rn-placeholder#60]
I couldn't find TypeScript [declaration file](https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html) of this library.

If you don't mind I'd make a PR to add the file directly to this repository, which is convenient for TypeScript users. Or I can put the file to [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) repository.

**Comment by mfrachet** [comment]
Hey!

I'm not really familiar with typescript ecosystem, so feel free to help us!

Thank you for this :blush:

**Comment by janhesters** [comment]
@yo7 Any news on this? This would bee awesome. If you don't do it, I'll do it!

**Comment by mfrachet** [comment]
I won't 😝 but if you're interested in, it would be great 👌

**Comment by yo7** [comment]
@janhesters Nice! (and sorry for being inactive).

I think it'd be convenient to directly put the definition to the repo, but maybe your choice is better in terms of maintainability🤔

**Comment by janhesters** [comment]
So I unfortunately don't have time to fork this and create a PR, but I'll leave the `index.d.ts` file here. @mfrachet Could you please add this to your package? Then we won't rely on the DefinitelyTyped repository.

```ts
// Type definitions for rn-placeholder 1.3
// Project: https://github.com/mfrachet/rn-placeholder
// Definitions by: Jan Hesters <https://github.com/janhesters>

import { SFC, Component, ComponentType, ReactNode } from "react";

export interface GenericPlaceholderProps {
    /**
     * How to animate the placeholder.
     */
    animate?: "fade" | "shine";

    /**
     * A custom animation represented by a React.Component;
     */
    customAnimate?: Component;

    /**
     * Whether the component should display its children (true) or the placeholder (false).
     */
    onReady?: boolean;
}

export interface LineProps extends GenericPlaceholderProps {
    /**
     * Color of the line.
     */
    color?: string;

    /**
     * Text size of the line.
     */
    textSize?: number;

    /**
     * Width of the line, percentage available;
     */
    width?: string;
}

export interface MediaProps extends GenericPlaceholderProps {
    /**
     * Color of the media shape.
     */
    color?: string;

    /**
     * Whether the media place holder be rounded or not.
     */
    hasRadius?: boolean;

    /**
     * The size of the media shape.
     */
    size?: number;
}

export interface ParagraphProps extends GenericPlaceholderProps {
    /**
     * Color of the lines.
     */
    color?: string;

    /**
     * Custom width of the first line.
     */
    firstLineWidth?: string;

    /**
     * Custom width of the last line.
     */
    lastLineWidth?: string;

    /**
     * Number of lines to display.
     */
    lineNumber: number;

    /**
     * Space between the lines.
     */
    lineSpacing?: number;

    /**
     * Text size of the lines.
     */
    textSize?: number;

    /**
     * Width of the lines.
     */
    width?: string;
}

export interface ImageContentProps extends ParagraphProps, MediaProps {
    /**
     * Color of the placeholder.
     */
    color?: string;

    /**
     * The position of the media.
     */
    position?: "left" | "right";
}

export interface BoxProps extends GenericPlaceholderProps {
    /**
     * The background color of the component.
     */
    color?: string;

    /**
     * Height of the component.
     */
    height?: number | string;

    /**
     * The border radius of the component.
     */
    radius?: number;

    /**
     * The width of the component.
     */
    width?: number | string;
}

interface Word {
    color: string;
    width: string;
}

export interface MultiWordProps extends GenericPlaceholderProps {
    words?: Word[];
    textSize?: number;
}

declare namespace Placeholder {
    /**
     * A React component for displaying a line placeholder.
     */
    const Line: SFC<LineProps>;

    /**
     * A React component for displaying a media placeholder.
     */
    const Media: SFC<MediaProps>;

    /**
     * A React component for displaying a paragraph placeholder.
     */
    const Paragraph: SFC<ParagraphProps>;

    /**
     * A React component for displaying an image content placeholder.
     */
    const ImageContent: SFC<ImageContentProps>;

    /**
     * A React component for displaying a box placeholder.
     */
    const Box: SFC<BoxProps>;

    /**
     * A React component for displaying a multi words placeholder.
     */
    const MultiWords: SFC<MultiWordProps>;

    const connect: () => ReactNode;
}

export default Placeholder;
```

**Comment by mfrachet** [comment]
I'll check with some mates to try that interface since I'm absolutely not confortable with TS 😅

**Comment by tomodian** [comment]
@janhesters you saved my day!

**Comment by mfrachet** [comment]
I try to put this inside a PR in the next days. Sorry for being that late 😓

**Comment by janhesters** [comment]
Made a pr: https://github.com/mfrachet/rn-placeholder/pull/66

**Comment by nirre7** [comment]
+1

**Comment by mfrachet** [comment]
https://github.com/mfrachet/rn-placeholder/pull/66 has been merged and published on the v1.3.3 version 😊

Thanks for your help (+ @janhesters 👍 ) and sorry for the long time 😞
